packages/e2e-test-utils-playwright/src/admin/create-new-post.js

```diff
@@ -30,26 +30,18 @@ export async function createNewPost( {
 
 	await this.visitAdminPage( 'post-new.php', query );
 
-	// Wait for both iframed and non-iframed canvas and resolve once the
-	// currently available one is ready. To make this work, we need an inner
-	// legacy canvas selector that is unavailable directly when the canvas is
-	// iframed.
-	await Promise.any( [
-		this.page.locator( '.wp-block-post-content' ).waitFor(),
-		this.page
-			.frameLocator( '[name=editor-canvas]' )
-			.locator( 'body > *' )
-			.first()
-			.waitFor(),
-	] );
```

Comment:
I think we should keep this. The missing lazy evaluation from `editor.canvas` wasn't the only reason for this safeguard.

## Why?
* Expectation that editor canvas is loaded when performing an action after the `createNewPost` call. Inherited from Puppeteer utils. It could count as a breaking change.
* Some of the test patterns don't use locators for creating blocks. See examples below.

## Examples

```js
// Insert a block using `editor` utils. Was flaky before https://github.com/WordPress/gutenberg/pull/51824.
await admin.createNewPost();
await editor.insertBlock( { name: 'core/heading' } );

// Add blocks using writing flow "shortcut":
await admin.createNewPost();
await page.keyboard.press( 'Enter' );
await page.keyboard.type( 'Content' );
```

The last pattern was very common in Puppeteer but became flaky after #48286 (before #51824), so we had to switch to `editor.canvas.click( 'role=button[name="Add default block"i]' )`.

Reply:
Believing that the entire editor is immediately ready upon invoking `callingNewPost()`, especially considering its dynamic nature, can be a risky assumption. Instead of making such assumptions, we should directly ensure that the resources we intend to use are available at the time of reference. Moreover, this check should occur right where the resource is accessed.

From the sample you shared, it seems more logical to adjust the `insertBlock()` function. Rather than having:

```js
async function insertBlock(
	this: Editor,
	blockRepresentation: BlockRepresentation
) {
	await this.page.evaluate( ( _blockRepresentation ) => {

		//...

		window.wp.data.dispatch( 'core/block-editor' ).insertBlock( block );
	}, blockRepresentation );
```

I propose this version:

```js
async function insertBlock(
	this: Editor,
	blockRepresentation: BlockRepresentation
) {
	await this.page.waitForFunction( ( _blockRepresentation ) => {
		if ( ! window?.wp?.data ) {
			return false;
		}

		//...

		window.wp.data.dispatch( 'core/block-editor' ).insertBlock( block );
	}, blockRepresentation );
```

In this alternative, the function waits until the `window.wp.data` resource is ready before proceeding. The immediate use of `evaluate()` presupposes the availability of `window.wp.data`, effectively moving the responsibility of ensuring its presence outside this function. To illustrate:

```js
await admin.createNewPost();
await editor.insertBlock( { name: 'core/heading' } );

await this.page.reload();
await editor.insertBlock( { name: 'core/heading' } );
```

If the responsibility to check resource availability lies within createNewPost, the second call could be unstable. However, if insertBlock self-validates that its required resources are ready, we ensure a more robust outcome. Does that perspective make sense?

Reply:
> Does that perspective make sense?

For me, it does. I don't feel a strong attachment to previous behavior 😄

As I mentioned, this changes the "expected" behavior of `createNewPost()`, and not everyone actively follows changes to the e2e test environment.

I like your suggestion for improving the `insertBlock` utility 🚀

Reply:
As for the second example you provided, it involves some ambiguity because we assume the cursor is in the correct spot, but we don't really know what that spot is. Instead of shifting the burden of ensuring _the cursor is where it should be_ to `createNewPost()` (which is not even the case, because we'd only wait for the editor body to be visible, which doesn't ensure cursor position!), I propose adding a step that explicitly waits for the cursor to be in the right spot:

```js
await admin.createNewPost();
await expect(
  editor.canvas.getByRole( 'textbox', { name: 'Add title' } )
).toBeFocused();

await page.keyboard.press( 'Enter' );
await page.keyboard.type( 'Content' );
```

Reply:
One DX benefit of removing this is it cleans up a little noise in the trace viewer that is caused by one of these locators timing out.